frontend/src/metabase/dashboard/dashboard.js

```diff
@@ -150,26 +146,49 @@ export const fetchCardData = createThunkAction(FETCH_CARD_DATA, function(card, d
             };
         }
 
-        let result = null;
+        const { dashboardId, dashboards, parameterValues, dashcardData } = getState().dashboard;
+        const dashboard = dashboards[dashboardId];
 
         // if we have a parameter, apply it to the card query before we execute
-        let { dashboardId } = getState().dashboard;
-        let { dashboards, parameterValues } = getState().dashboard;
+        const datasetQuery = applyParameters(card, dashboard.parameters, parameterValues, dashcard && dashcard.parameter_mappings);
 
-        let dashboard = dashboards[dashboardId];
+        if (!reload) {
+            // if reload not set, check to see if the last result has the same query dict and return that
+            const lastResult = i.getIn(dashcardData, [dashcard.id, card.id]);
+            // "constraints" is added by the backend, remove it when comparing
+            if (lastResult && angular.equals(_.omit(lastResult.json_query, "constraints"), datasetQuery)) {
```

Comment:
`angular.equals` O.o seems like there should be a way to determine equivalent values without using an angular function.